app/models/miq_report_result/result_set_operations.rb

```diff
@@ -6,16 +6,45 @@ def apply_limit_and_offset(results, options)
       results.slice(options['offset'].to_i, options['limit'].to_i) || []
     end
 
+    FILTER_PARAMS_MAX_COUNT = 20
+
+    def filter_field(parameter, param_number)
+      filter_param_suffix = param_number.zero? ? '' : "_#{param_number}"
+      "#{parameter}#{filter_param_suffix}"
+    end
+
+    def filtering_enabled?(options)
+      options.key?(:filter_column) || options.key?(:filter_column_1)
+    end
+
+    def filter_options(options)
+      filter_params = {}
+      (0...FILTER_PARAMS_MAX_COUNT).each_with_object({}) do |param_number, _|
```

Comment:
```diff
- filter_params = {}
```

```suggestion
      (0...FILTER_PARAMS_MAX_COUNT).each_with_object({}) do |param_number, filter_params|
```

```diff
- break unless param_value
+ break(filter_params) unless param_value
```


```diff
- filter_params #last line in method
```

Reply:
that's cool, I didn't know about `break(returnthis)`

thanks 👍